src/sagemaker/utils.py

```diff
@@ -1655,3 +1657,54 @@ def deep_override_dict(
     )
     flattened_dict1.update(flattened_dict2)
     return unflatten_dict(flattened_dict1) if flattened_dict1 else {}
+
+
+def get_instance_rate_per_hour(
+    instance_type: str,
+    region: str,
+    pricing_client: boto3.client = boto3.client("pricing", region_name="us-east-1"),
```

Comment:
does it have to be us-east-1? should we use the sagemaker session to derive the region/client?

Reply:
It can only one of the following
* https://api.pricing.us-east-1.amazonaws.com
* https://api.pricing.ap-south-1.amazonaws.com
* https://api.pricing.eu-central-1.amazonaws.com
https://docs.aws.amazon.com/aws-cost-management/latest/APIReference/Welcome.html#Welcome_AWS_Price_List_Service

The actual region query happens at `{"Type": "TERM_MATCH", "Field": "regionCode", "Value": region},`